### PR TITLE
[MAINT] Implement deprecation warnings-as-errors in prerelease test using pytest fixture

### LINKED ISSUE #3541: Suggestion to use warnings-as-errors with pytest in prerelease CI
> There is at least one test that would fail under the original bug conditions.

To find stuff like this in MNE we test with warnings-as-errors in pytest, and use a prerelease version of dependencies like `matplotlib`. It helps stay ahead of the curve a bit. But it would be a big change here...

I could instead change your pytest config to at least error on matplotlib deprecations, then you'd at least see them when they cut new releases. Not sure how many other failures this would trigger, it took us a bit to get to 3.7-deprecation-free compat over in MNE...

_Originally posted by @larsoner in https://github.com/nilearn/nilearn/issues/3535#issuecomment-1442270288_
            

Making sure this doesn't get lost. This would be added to our prerelease job in the testing workflow https://github.com/nilearn/nilearn/blob/c7aa9f74ec0b13f4946921d0b4b921e81a8639f1/.github/workflows/testing.yml#L106-L114

### PR DESCRIPTION
**Pull request by ymzayek**
- Closes #3541 and maybe #1737

Changes proposed in this pull request:

-
-

### CODE REVIEW

**Comment by ymzayek** [comment]
TODO: set this up as a separate workflow that runs once a month or so like `detect_test_pollution` workflow. Definitely don't want to burden contributors with a maintenance job.

File: .github/workflows/check_deprecations.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#            │ ┌───────────── hour (0 - 23)
+#            │ │ ┌───────────── day of the month (1 - 31)
+#            │ │ │ ┌───────────── month (1 - 12 or JAN-DEC)
+#            │ │ │ │ ┌───────────── day of the week (0 - 6 or SUN-SAT)
+#            │ │ │ │ │
+#            │ │ │ │ │
+#            │ │ │ │ │
+#            * * * * *
+    schedule:
+    -   cron: 0 0 25 * * # on the 25th of every monday
```
Comment (bthirion) [inline .github/workflows/check_deprecations.yml:25]:
```suggestion
    -   cron: 0 0 25 * * # on the 25th of every month
```

**Review by bthirion — COMMENTED** [review]
LGTM otherwise.

File: .github/workflows/check_deprecations.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        defaults:
+            run:
+                shell: bash
+        steps:
+        -   uses: actions/checkout@v4
+
+        -   name: Setup python
+            uses: actions/setup-python@v4
+            with:
+                python-version: 3.11
```
Comment (Remi-Gau) [inline .github/workflows/check_deprecations.yml:46]:
```suggestion
                python-version: 3.12
```

**Review by Remi-Gau — APPROVED** [review]
Good with me.
I guess we should trigger it after merging to get a baseline

**Comment by ymzayek** [comment]
Let's merge it and then trigger it manually.

**Comment by Remi-Gau** [comment]
> Let's merge it and then trigger it manually.

and everything went fine

**Pull request merged**